### DISCUSSION TITLE
Why this rediculous amount of code to draw a simple rectangle?

### DISCUSSION THREAD

**Discussion by RoelRoel** [apexcharts/apexcharts.js#2717] [Q&A]
I have a bar chart in my application an I was examinating the DOM in the developer tools. I was a bit shocked by the elements that Apex Chart generate. This is only one bar at the bar chart. It draws a rectangle:

`<path id="SvgjsPath4332" d="M 40.25980392156863 286.30301353708904L 40.25980392156863 90.90120679802578Q 40.25980392156863 90.90120679802578 40.25980392156863 90.90120679802578L 53.367647058823536 90.90120679802578Q 53.367647058823536 90.90120679802578 53.367647058823536 90.90120679802578L 53.367647058823536 90.90120679802578L 53.367647058823536 286.30301353708904L 53.367647058823536 286.30301353708904z" fill="rgba(57,143,83,0.85)" fill-opacity="1" stroke-opacity="1" stroke-linecap="round" stroke-width="0" stroke-dasharray="0" class="apexcharts-bar-area" index="0" clip-path="url(#gridRectMaskqmqbjzgq)" pathTo="M 40.25980392156863 286.30301353708904L 40.25980392156863 90.90120679802578Q 40.25980392156863 90.90120679802578 40.25980392156863 90.90120679802578L 53.367647058823536 90.90120679802578Q 53.367647058823536 90.90120679802578 53.367647058823536 90.90120679802578L 53.367647058823536 90.90120679802578L 53.367647058823536 286.30301353708904L 53.367647058823536 286.30301353708904z" pathFrom="M 40.25980392156863 286.30301353708904L 40.25980392156863 286.30301353708904L 53.367647058823536 286.30301353708904L 53.367647058823536 286.30301353708904L 53.367647058823536 286.30301353708904L 53.367647058823536 286.30301353708904L 53.367647058823536 286.30301353708904L 40.25980392156863 286.30301353708904" cy="90.90120679802578" cx="58.985294117647065" j="2" val="68.25" barHeight="195.40180673906326" barWidth="13.107843137254902"></path>`

All this is used to only render this one blue bar in this image below:
![image](https://user-images.githubusercontent.com/531248/138467554-34aa6051-3c05-4e76-93e5-d9b31e5a8408.png)

If I analyse it a bit the path has the 4 coordinates needed to draw it 4 times * 3. 
And why does it use the complete path as a class? This must be very slow for the client if you have a lot of bars....

**Comment by junedchhipa** [comment]
Most of the code is just data-attributes (pathFrom, pathTo) which are used for the next re-render. The reason `path` is used instead of 
 `rect` is due to features like border-radius.

**Comment by RoelRoel** [comment]
Ok, but the path alone is drawing this rectangle multiple times. The
coordinates of the corners seem to be repeated multiple times.

Op vr 22 okt. 2021 18:14 schreef Juned Chhipa ***@***.***>:
